tools/sql_extraction/src/main/kotlin/FileListExpander.kt

```diff
@@ -4,7 +4,20 @@ import java.nio.file.Files
 import java.nio.file.Path
 import java.util.stream.Collectors
 
+/**
+ * Expands a collection of file and directory paths to a list of all applicable file paths
+ */
 class FileListExpander {
+    /**
+     * Converts a collection of file and directory paths to all applicable file paths
+     *
+     * @param[dirs] File and directory paths to consider
+     * @param[recursive] True of visit subdirectories recursively,
+     *     false to only visit the top-level directories
+     * @param[includes] List of GLOB patterns to limit added output file paths
+     * @param[excludes] List of GLOB patterns to filter out output file paths
+     * @return Expanded and filtered collection of file paths to analyze
```

Comment:
This is OK today. But pls consider what happens when there are many files to be returned. It's better in that case to use lazy evaluation and return the result in a stream.

Reply:
Wow good idea!